### ISSUE TITLE
List reflex fields of CM fields

### ISSUE THREAD

**Issue by AndrewVSutherland** [LMFDB/lmfdb#4902]
For number fields that are CM fields, it would be nice to list the possible CM types (choice of embedding for each complex place) and the corresponding reflex fields (even just a list of the reflex fields would be useful, they should all be in the database).

**Comment by rbommel** [comment]
Note to anyone who wants to think about this in the future (maybe myself): some fields to start trying to compute this for could be https://beta.lmfdb.org/NumberField/14.0.121781095161992523.1 or https://beta.lmfdb.org/NumberField/36.0.2167466163148151967744828835955985745949336362631879785820717056.1.

**Comment by rbommel** [comment]
Today, we discussed only computing this for CM fields of degree at most 12, and to only compute equations for the reflex field is that field itself has degree at most 36. We also discussed computing whether or not the reflex field is CM, but after a bit more consideration, I think they are always CM.

**Comment by AndrewVSutherland** [comment]
The reflex field of a CM type is a CM field, see Proposition 1.18 of [Milne's notes](https://www.jmilne.org/math/CourseNotes/CM.pdf).

The reflex field of a Shimura variety can be totally real or CM, but here we are talking about reflex fields of CM types (associated to CM abelian varieties).

**Comment by JohnCremona** [comment]
I am working with Marco Streng and Pinar Kilicer at COUNT to display the CM types for each reflex field, not just the multiplicities.  For this we are adding 2 columns to nf_fields to hold the real and imaginary parts of the embeddings of the field generator (only for the CM fields of degree up to 12 at first), and one column to nf_fields_reflex to hold a list of CM-types, each encoded as a string like "+-+-".  This will require more computation by @rbommel.

**Comment by jwj61** [comment]
Is this done and can be closed?

**Comment by JohnCremona** [comment]
> Is this done and can be closed?

No.  Work on it started at Luminy in Feb/March, and a branch exists with a table with a few test rows, but the work still needs to be done to do the remaining computations.

**Comment by jwj61** [comment]
The table has almost 3 million entries.

**Comment by JohnCremona** [comment]
I'm clearly out of date on this one!

**Comment by rbommel** [comment]
Whether this is still the same issue or not, we are working on an upgrade for this, including complex embeddings of the reflex fields, Galois orbits, and reflex reflex fields (which are 'fields of definition' for the CM types). For more info, I refer to the discussion on the COUNT Zulip.

**Comment by roed314** [comment]
Checking in on this issue: has more progress been made since @rbommel's last update?

**Comment by rbommel** [comment]
No, there has been a Magma bug for a while that halted this, but it may be the case now that it has been fixed, but I'd have to look into that again.
